### ISSUE TITLE
Manual Handover

### ISSUE THREAD

**Issue by lhupalo** [wineslab/ns-o-ran-ns3-mmwave#7]
Hi there,

I'm trying to perform manual handovers, but I'm getting the following error:

`
assert failed. cond="m_x2InterfaceSockets.find (params.targetCellId) != m_x2InterfaceSockets.end ()", msg="Missing infos for targetCellId = 2", +0.401000003s 3 file=../src/lte/model/epc-x2.cc, line=668
terminate called without an active exception
`
Sometimes it work, but sometimes it doesn't work and shows the message above. I would guess that this is occurring due to some configuration that I missed, or maybe the base stations are not being initialized correctly, but I couldn't figure out.

I'm doing this procedure exactly to the code in `LteEnbNetDevice` through the `Simulator::Schedule (Seconds (0.0), &LteEnbRrc::PerformHandoverToTargetCell, rrc_pointer, imsi, targetCellId);`. The only difference is that I'm not reading from a csv file, but parsing the values directly in the function.

Please, let me know if I am doing something wrong. 

Best Regards,
Luiz

**Comment by Thecave3** [comment]
Hi!

I might need some additional context to help you. 
Can you please share the points in the code that you have changed?

**Comment by lhupalo** [comment]
Hello!

Here is the function that I built:

```
void perform_handover_to_target_cell(uint64_t imsi, uint16_t targetCellId, NetDeviceContainer lteEnbDevs)
{
  Ptr<NetDevice> lte_dev = lteEnbDevs.Get(0);
  Ptr<LteEnbRrc> rrc_pointer = lte_dev->GetObject<LteEnbNetDevice>()->GetRrc();

  Simulator::Schedule (Seconds (0.0), &LteEnbRrc::PerformHandoverToTargetCell, rrc_pointer, imsi, targetCellId);
}
```
Here, I'm calling this function on the gym `MyExecuteActions()`. My scenario is composed of one LTE and 3 gNBs. From what I've learned from the paper, in this dual connectivity scheme the UEs are connected to a LTE and to a gNB at the same time, but we can only perform handovers between the gNBs stations (and this handover between the gNBs must be triggered through the LTE NetDevice, which is the entity that hold the control these type of procedures). Am I right to think this way?

If yes, when a HO is triggered, I thought that calling directly the RRC pointer of the LTE NetDevice I would be able to switch between the gNBs. As I said, it worked for some steps;  if the conditions are satisfied (UE initialized and attached to some gNB, UE not performing handover at the moment of the trigger, target BS with the X2 interface socket initialized properly) it works. But sometimes, the last mentioned condition falls apart, and the error that I first mentioned here on the issue appears.

The rest of the code is the same as every ns-3 simulation. I didn't try the handover method using the csv file because I wanted to keep all inside ns-3, but now I'm considering to try it and see if its a problem of the rest of the code or in my handover function.

Thanks,
Luiz
